modelon/impact/client/entities/workspace.py

```diff
@@ -407,6 +413,20 @@ def size(self) -> float:
     def definition(self) -> WorkspaceDefinition:
         return self._workspace_definition
 
+    @property
+    def name(self) -> str:
```

Comment:
This property becomes a bit weird since setting the name results in a remote call and definition re-fetch while reading returns client-side value and does not make additional server request. This implies:
```
workspace = client.get_workspace(id)
same_workspace = client.get_workspace(id)
workspace.name = "new_name"
assert workspace.name != same_workspace.name
same_workspace = client.get_workspace(id)
assert workspace.name == same_workspace.name
```
Therefore, I this I'd prefer a "rename" method with docstring clearly stating that it does a remote call.

Reply:
I agree. I have now added a explicit rename method. However I think we should hide details like "the api does a remote" from the user. Instead I think we should always try to fetch the definition when workspace.definition is called. If we do this we can avoid such weird behavior like the one you pointed out. and always be confident of being in sync with the remote? I think the cost of doing the call to remote always isn't significant. Thoughts? :)